File: fixtures/typescript/env-secret/positive.ts
// Calls hosted APIs.

// must fire: declarator value
const apiKey = process.env.OPENAI_API_KEY;

function config() {
  return {
    // must fire: object-literal value
    key: process.env.ANTHROPIC_API_KEY,
    // must fire: operand of ||
    fallback: process.env.REPLICATE_API_TOKEN || null,
  };
}

function signer() {
  // must fire: return position
  return process.env.DEPLOYER_PRIVATE_KEY;
}

module.exports = { apiKey, config, signer };

// must fire: TypeScript's non-null assertion wraps the access, which an
// enumeration of parent contexts cannot see through. A real corpus miss.
const clientSecret = process.env.CLIENT_SECRET!;
const tenantToken = process.env.TENANT_TOKEN as string;

File: fixtures/python/env-secret/positive.py
"""Calls a hosted model."""

import os


def client():
    # must fire: os.getenv with a secret-bearing name
    key = os.getenv("OPENAI_API_KEY")
    # must fire: the .get form, same act
    fallback = os.environ.get("ANTHROPIC_API_KEY")
    return key or fallback


def signer():
    # must fire: a private key is a secret whatever the prefix
    return os.getenv("DEPLOYER_PRIVATE_KEY")
